exploration/open-close-expressions.md

```diff
@@ -220,3 +221,50 @@ Rendered as React this could become:
 _What other solutions are available?_
 _How do they compare against the requirements?_
 _What other properties they have?_
+
+### HTML-like syntax
+
+The goal of this solution is to avoid adding new sigils to the syntax.
+Instead, it leverages the familiarity of the `foo`...`/foo` idiom,
+inspired by HTML and BBCode.
+
+This solution consists of adding new placeholder syntax:
+`{foo}`, `{/foo}`, and `{foo/}`.
+
+```
+This is {html:strong}bold{/html:strong} and this is {html:img alt=|an image|/}.
+```
+
+Markup names are *effectively namespaced* due to their not using any sigils;
+they are distinct from `$variables`, `:functions`, and `|literals|`.
+
+> [!NOTE]
+> This requires dropping unquoted literals as operands,
+> so that `{foo}` is not parsed as `{|foo|}`.
```

Comment:
I think this is relatively difficult to contemplate. We will want unquoted literals for keys. Is unquoted literals for operands different enough a use case?

Reply:
Hmm, I think I agree. Expressions basically break down into
* literal with optional function
* variable with optional function
* operand-free function
* spannable open/close

If both variables and functions use a sigil, then I think spannables should as well, leaving literals as the sole sigil-free expression contents (and with consistent quoting requirements regardless of where they appear).

Reply:
I filed #518 to discuss this further.